### PR TITLE
fix: cluster connection audit

### PR DESCRIPTION
**Pull request by sherpalden**
### Commit Message

```bash
type: commit message
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
version: <log entry>
```

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have run the unit tests
- [ ] I only have one commit (if not, squash them into one commit).
- [x] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

File: contracts/evm/contracts/adapters/ClusterConnection.sol [hunk 1]
```diff
@@ -141,42 +146,65 @@ contract ClusterConnection is Initializable, IConnection {
         string memory srcNetwork,
         uint256 _connSn,
         bytes calldata _msg,
+        string memory dstNetwork,
```
Comment (AntonAndell) [inline contracts/evm/contracts/adapters/ClusterConnection.sol:149]: This is not needed as a parameter but should preferably be taken from the contract itself

File: contracts/evm/test/adapters/ClusterConnection.t.sol [hunk 2]
```diff
@@ -341,12 +387,20 @@ contract ClusterConnectionTest is Test {
...
+        uint256 _connSn,
+        bytes memory _msg,
+        string memory dstNetwork
+    ) internal pure returns (bytes32) {
+        bytes memory rlp = abi
+            .encodePacked(
+                srcNetwork.encodeString(),
+                _connSn.encodeUint(),
+                _msg.encodeBytes(),
+                srcNetwork.encodeString()
```
Comment (gcranju) [inline contracts/evm/test/adapters/ClusterConnection.t.sol:401]:
```suggestion
                dstNetwork.encodeString()
```

File: contracts/evm/contracts/adapters/ClusterConnection.sol [hunk 3]
```diff
@@ -268,24 +298,34 @@ contract ClusterConnection is Initializable, IConnection {
...
+        uint256 _connSn,
+        bytes calldata _msg,
+        string memory dstNetwork
+    ) internal pure returns (bytes32) {
+        bytes memory rlp = abi
+            .encodePacked(
+                srcNetwork.encodeString(),
+                _connSn.encodeUint(),
+                _msg.encodeBytes(),
+                srcNetwork.encodeString()
```
Comment (gcranju) [inline contracts/evm/contracts/adapters/ClusterConnection.sol:320]:
```suggestion
                dstNetwork.encodeString()
```

**Review by gcranju — CHANGES_REQUESTED** [review]

File: contracts/evm/contracts/adapters/ClusterConnection.sol (hunk 1 above)
Comment (sherpalden) [reply contracts/evm/contracts/adapters/ClusterConnection.sol]: Updated

**Review by ibrizsabin — APPROVED** [review]

**Review by gcranju — APPROVED** [review]

**Pull request merged**